From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Giuliano Procida <gprocida@google.com>
Date: Fri, 18 Sep 2020 11:00:54 +0100
Subject: abg-corpus.cc: report architecture discrepancies

If there ever is a discrepancy between the architectures of the
corpuses of a corpus group, libabigail will just abort with an
assertion, if enabled. However, this is a case of invalid input and
the cause should be reported to the user.

	* src/abg-corpus.cc (corpus_group::add_corpus): Report
	architecture discrepancies.

Signed-off-by: Giuliano Procida <gprocida@google.com>
Signed-off-by: Matthias Maennich <maennich@google.com>
---
 src/abg-corpus.cc | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/src/abg-corpus.cc b/src/abg-corpus.cc
index 1f72904d137f..67f878dd6266 100644
--- a/src/abg-corpus.cc
+++ b/src/abg-corpus.cc
@@ -1647,8 +1647,13 @@ corpus_group::add_corpus(const corpus_sptr& corp)
     corp_arch = corp->get_architecture_name();
   if (cur_arch.empty())
     set_architecture_name(corp_arch);
-  else
-    ABG_ASSERT(cur_arch == corp_arch);
+  else if (cur_arch != corp_arch)
+    {
+      std::cerr << "corpus '" << corp->get_path() << "'"
+		<< " has architecture '" << corp_arch << "'"
+		<< " but expected '" << cur_arch << "'\n";
+      ABG_ASSERT_NOT_REACHED;
+    }
 
   priv_->corpora.push_back(corp);
   corp->set_group(this);
